pkg/manager/impl/artifact_manager.go

```diff
@@ -193,13 +211,74 @@ func (m *artifactManager) GetArtifact(ctx context.Context, request datacatalog.G
 			Value: value,
 		}
 	}
-	artifact.Data = artifactDataList
 
-	logger.Debugf(ctx, "Retrieved artifact dataset %v, id: %v", artifact.Dataset, artifact.Id)
-	m.systemMetrics.getSuccessCounter.Inc(ctx)
-	return &datacatalog.GetArtifactResponse{
-		Artifact: &artifact,
-	}, nil
+	return artifactDataList, nil
+}
+
+func (m *artifactManager) ListArtifacts(ctx context.Context, request datacatalog.ListArtifactsRequest) (*datacatalog.ListArtifactsResponse, error) {
+	err := validators.ValidateListArtifactRequest(&request)
+	if err != nil {
+		logger.Warningf(ctx, "Invalid list artifact request %v, err: %v", request, err)
+		m.systemMetrics.validationErrorCounter.Inc(ctx)
+		return nil, err
+	}
+
+	// Verify the dataset exists before listing artifacts
+	datasetKey := transformers.FromDatasetID(*request.Dataset)
+	dataset, err := m.repo.DatasetRepo().Get(ctx, datasetKey)
+	if err != nil {
+		logger.Warnf(ctx, "Failed to get dataset for listing artifacts %v, err: %v", datasetKey, err)
+		m.systemMetrics.listFailureCounter.Inc(ctx)
+		return nil, err
+	}
+
+	// Get the list inputs
+	listInput, err := transformers.FilterToListInput(ctx, common.Artifact, request.GetFilter())
+	if err != nil {
+		logger.Warningf(ctx, "Invalid list artifact request %v, err: %v", request, err)
+		m.systemMetrics.validationErrorCounter.Inc(ctx)
+		return nil, err
+	}
+
+	err = transformers.ApplyPagination(request.Pagination, &listInput)
+	if err != nil {
+		logger.Warningf(ctx, "Invalid pagination options in list artifact request %v, err: %v", request, err)
+		m.systemMetrics.validationErrorCounter.Inc(ctx)
+		return nil, err
+	}
+
+	// Perform the list with the dataset and listInput filters
+	artifactModels, err := m.repo.ArtifactRepo().List(ctx, dataset.DatasetKey, listInput)
+	if err != nil {
+		logger.Errorf(ctx, "Unable to list Artifacts err: %v", err)
+		m.systemMetrics.listFailureCounter.Inc(ctx)
+		return nil, err
+	}
+
+	// convert returned models into entity list
+	artifactsList, err := transformers.FromArtifactModels(artifactModels)
+	if err != nil {
+		logger.Errorf(ctx, "Unable to transform Artifacts %+v err: %v", artifactModels, err)
+		m.systemMetrics.listFailureCounter.Inc(ctx)
+		return nil, err
+	}
+
+	// attach artifactData to artifactList
+	for i, artifact := range artifactsList {
```

Comment:
good question, the call below this `m.getArtifactDataList` retrieves the files from S3 and unmarshals them. So I don't think it belongs in transformers since it isn't just thin logic. That being said I think you brought this up in the last review for `getArtifact`, so I will definitely add a comment here and in `getArtifact` explaining that